SCORER.md

```diff
@@ -21,11 +25,175 @@ You may need to build your own scorer - your own _language model_ if:
+ 392 -rwxr-xr-x 1 root root  399790 Feb  9 10:43 test.tsv
+ 456 -rwxr-xr-x 1 root root  465258 Feb  9 10:43 train.tsv
+1848 -rwxr-xr-x 1 root root 1889606 Feb  9 10:43 validated.tsv
+```
+
+The `indonesian-sentences.txt` file is stored on the local filesystem in the `deepspeech-data` directory so that the Docker container can access it.
+
+```
+~/deepspeech-data$ ls | grep indonesian-sentences
+ 476 -rw-rw-r--  1 root root  483481 Feb 24 19:02 indonesian-sentences.txt
+```
+
+#### Using `generate_lm.py` to create `lm.binary` and `vocab-500000.txt` files
+
+We then use `generate_lm.py` script that comes with DeepSpeech to create a [_trie file_](https://en.wikipedia.org/wiki/Trie). The _trie file_ represents associations between words, so that during training, words that are more closely associated together are more likely to be transcribed by DeepSpeech.
+
+The _trie file_ is produced using a software package called [KenLM](https://kheafield.com/code/kenlm/). KenLM is designed to create large language models that are able to be filtered and queried easily.
+
+First, create a directory in `deepspeech-data` directory to store your `lm.binary` and `vocab-500000.txt` files:
+
+```
+deepspeech-data$ mkdir indonesian-scorer
+```
+
+Then, use the `generate_lm.py` script as follows:
+
+```
+cd data/lm
+python3 generate_lm.py \
+  --input_txt /DeepSpeech/deepspeech-data/indonesian-sentences.txt \
+  --output_dir /DeepSpeech/deepspeech-data/indonesian-scorer \
+  --top_k 500000 --kenlm_bins /DeepSpeech/native_client/kenlm/build/bin/ \
+  --arpa_order 5 --max_arpa_memory "85%" --arpa_prune "0|0|1" \
+  --binary_a_bits 255 --binary_q_bits 8 --binary_type trie
+```
+
+_Note: the `/DeepSpeech/native_client/kenlm/build/bin/` is the path to the binary files for `kenlm`. If you are using the Docker image and container (explained on the [environment page of the PlayBook](ENVIRONMENT.md)), then `/DeepSpeech/native_client/kenlm/build/bin/` is the correct path to use. If you are not using the Docker environment, your path may vary._
+
+You should now have a `lm.binary` and `vocab-500000.txt` file in your `indonesian-scorer` directory:
+
+```
+deepspeech-data$ ls indonesian-scorer/
+total 1184
+  4 drwxrwxr-x 2 root   root   4096 Feb 25 23:13 ./
+  4 drwxrwxr-x 5 root   root   4096 Feb 26 09:24 ../
+488 -rw-r--r-- 1 root   root      499594 Feb 24 19:05 lm.binary
+ 52 -rw-r--r-- 1 root   root       51178 Feb 24 19:05 vocab-500000.txt
+```
+
+#### Generating a `kenlm.scorer` file from `generate_scorer_package`
+
+Next, we need to install the `native_client` package, which contains the `generate_scorer_package`. This is _not_ pre-built into the DeepSpeech Docker image.
+
+First, download the relevant `native client` package from the [DeepSpeech GitHub releases page](https://github.com/mozilla/DeepSpeech/releases/tag/v0.9.3) into the `data/lm` directory. The Docker image uses Ubuntu Linux, so you should use the `native_client.amd64.cuda.linux.tar.xz` package.
```

Comment:
Might be useful mentionning that `generate_scorer_package` is usable on all platforms we support so people can generate scorers on-device (e.g., on an android device, or rpi3, ...)

Reply:
Great feedback, thank you! Have added a paragraph to this effect.